src/common/src/types/struct_type.rs

```diff
@@ -70,3 +71,28 @@ impl Display for StructType {
         }
     }
 }
+
+impl FromStr for StructType {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s == "record" {
```

Comment:
What format is the `str` in? Is it SQL representation?

Reply:
It is almost auto-generated by Copilot from the above Display code. 😄